test/js-api/jsapi.js

```diff
@@ -661,6 +664,112 @@ test(() => {
     assertThrows(() => tbl.grow(1), Error);
 }, "'WebAssembly.Table.prototype.grow' method");
 
+test(() => {
+    const globalDesc = Object.getOwnPropertyDescriptor(WebAssembly, 'Global');
+    assert_equals(typeof globalDesc.value, "function");
+    assert_equals(globalDesc.writable, true);
+    assert_equals(globalDesc.enumerable, false);
+    assert_equals(globalDesc.configurable, true);
+    Global = WebAssembly.Global;
+}, "'WebAssembly.Global' data property");
+
+test(() => {
+    const globalDesc = Object.getOwnPropertyDescriptor(WebAssembly, 'Global');
+    assert_equals(Global, globalDesc.value);
+    assert_equals(Global.length, 1);
```

Comment:
Length should be 2 as it takes a `descriptor` and `value` now

Reply:
But value is optional, so it should be 1 right?

Reply:
> According to ECMA
> 19.2.4.1length
The value of the length property is an integer that indicates the typical number of arguments expected by the function

I admit the wording is vague, but by looking at `parseInt.length` which is `2` and the second argument is optional like in this case.
I would say the typical number of arguments includes optional arguments

Reply:
It seems that it is specified here: https://tc39.github.io/ecma262/#sec-function-definitions-static-semantics-expectedargumentcount

In particular, the expected argument count stops at the first formal parameter that has an initializer:

>  If HasInitializer of FormalParameterList is true or HasInitializer of FormalParameter is true, return count.

ExpectedArgumentCount is then used to define the length of an initialized function: https://tc39.github.io/ecma262/#sec-functioninitialize

You can see this with a user-defined function:
```
(function(a, b=2){}).length // 1
```

It seems that `parseInt` is not defined as having a default formal parameter, which explains why it is length 2: https://tc39.github.io/ecma262/#sec-parseint-string-radix

If you look at a function with a default argument like `Array.prototype.reduce(callbackFn [,initialValue])`, the length is 1: https://tc39.github.io/ecma262/#sec-array.prototype.reduce

It actually doesn't define the length here, so perhaps it falls back to the general rule in `FunctionInitialize`?

Any JS language lawyer want to help us out here? :-)

Reply:
I think your argument with user-defined function convinced me it should be length 1.